--- a/amprd.c
+++ b/amprd.c
@@ -1,3 +1,9 @@
+#include <netinet/udp.h>
+#include <arpa/inet.h>
+
+#ifndef IPPORT_ROUTESERVER
+#define IPPORT_ROUTESERVER 520
+#endif
 #include <errno.h>
 #include <fcntl.h>
 #include <net/if.h>
@@ -554,8 +560,8 @@
-    if (uhdr->source == htons(IPPORT_ROUTESERVER) &&
-        uhdr->dest   == htons(IPPORT_ROUTESERVER))
+    if (uhdr->uh_sport == htons(IPPORT_ROUTESERVER) &&
+        uhdr->uh_dport == htons(IPPORT_ROUTESERVER))
 	return;
 
 	/* process datagram */
